testSettings.py

```diff
@@ -1,56 +1,49 @@
+import testConfig
+
 from tftbase import ConnectionMode
 from tftbase import NodeLocation
 from tftbase import PodInfo
 from tftbase import PodType
 from tftbase import TestCaseType
 from tftbase import TestMetadata
-from tftbase import TestType
 
 
 class TestSettings:
     """TestSettings will handle determining the logic require to configure the client/server for a given test"""
 
     def __init__(
         self,
-        connection_name: str,
+        connection: testConfig.ConfConnection,
         test_case_id: TestCaseType,
-        node_server_name: str,
-        node_client_name: str,
-        server_pod_type: PodType,
-        client_pod_type: PodType,
-        server_default_network: str,
-        client_default_network: str,
```

Comment:
I am confused where is "default_network" set or more specifically "conf_client.default_network"

Reply:
`conf_client` is the `testConfig.ConfClient` type, which is an object to keep the data from the test YAML, like
```
tft:
  - name: test1
    connections:
      - name: connection1
        client:
          - name: client1 # <<< HERE
            default-network: default/default
            sriov: True
```

This is parsed in testClient.py, see
```python
        default_network = "default/default"
        v = vdict.pop("default-network", None)
        if v is not None:
            if not isinstance(v, str):
                raise ValueError(f'"{yamlpath}.name": expects a string but got {name}')
            default_network = v
```

Reply:
basically, search for `default_network` or `"default-network"` in `testConfig.py`. It comes from there.

And since all of `TestConfig` is immutable, it *only* gets initialized while creating the `TestConfig` instance, which happens early in `main.py` when we parse the YAML.

Reply:
Thanks, I think those changes were made in another commit that was why I was confused.